packages/nocodb/src/modules/datas/helpers.ts

```diff
@@ -268,11 +268,15 @@ export async function getDbRows(param: {
       break;
     }
 
+    const needSerializeColumns = view.model.columns.filter((column) => {
```

Comment:
@willnewii can you provide a scenario in detail so that I could test the difference? and why we need to filter `view.model.columns` based on the first row? Supposedly if a field is hidden, then it won't be in `view.model.columns`. In this case, after filtering, `needSerializeColumns` should be same as `view.model.columns`.